### PR TITLE
issue #199-Added side menu for react admin 

### PR DESCRIPTION
**Pull request by felizj17**
Signed-off-by: Juan Feliz <felizj17@gmail.com>

Contributes to #199 

## What did you do?
created the side menu for react admin.
## Why did you do it?
it was needed to more easily route to pages
## How have you tested it?
tested available routes.  I have not created test for it yet.
## Were docs updated if needed?

- [ ] No
- [ ] Yes
- [x] N/A

#### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new console logs
- [ ] Fixes entire issue
- [x] Partial fix for issue

### CODE REVIEW

**Comment by felizj17** [comment]
I moved the BrowserRouter into the index.js so that the side menu can close when routing to a new location i hope thats alright

File: server/react-admin/src/AdminApp.js [hunk 1]
```diff
@@ -1,114 +1,109 @@
...
+import SideNav from './sidemenu/SideNav'
 import AdminHomeRoute from './adminHomeRoute/AdminHomeRoute'
-import CategoryRoute from './categories/CategoryRoute'
-import CategoryForm from './categories/CategoryForm'
 import PublicationRoute from './publications/PublicationRoute'
 import PublicationForm from './publications/PublicationForm'
 import VideoTestimonialRoute from './videoTestimonials/VideoTestimonialRoute'
 import VideoTestimonialForm from './videoTestimonials/VideoTestimonialForm'
 import AdvocacyGroupsRoute from './advocacyGroups/AdvocacyGroupsRoute'
 import AdvocacyGroupsForm from './advocacyGroups/AdvocacyGroupsForm'
```
Comment (markstur) [inline server/react-admin/src/AdminApp.js:11]: Needs to be updated so it doesn't delete the routes to category, levels, geography.

**Review by markstur — COMMENTED** [review]
Not sure if this is a draft or a real pull request.  It needs some updating so it doesn't lose recent work. Juan, could you mark it as a draft and when it is updated and ready for approve/merge remember to un-draft it?

**Comment by markstur** [comment]
This looks great. I'd like to get this on main so we have the functionality, but first I'm hoping @ed-snodgrass has a chance to look in particular to make sure he agrees with the BrowserRouter rearranging. So we don't go back and forth on something like that (unless back and forth good iterations).  I don't see a problem with it.

**Comment by ed-snodgrass** [comment]
@markstur I'm curious if we could simplify the side menu a bit and just have the links to the individual components? The table components themselves have the add/edit options.

Something like:
image

File: server/react-admin/src/AdminApp.js [hunk 2]
```diff
@@ -16,134 +18,152 @@ import LevelsForm from './levels/LevelsForm'
 import GeographyRoute from './geographies/GeographyRoute'
 import GeographyForm from './geographies/GeographyForm'
 
+import './adminapp.scss'
+const drawerWidth = 280
+
 function AdminApp() {
+  const location = useLocation()
+  const [open, setOpen] = React.useState(true)
```
Comment (ed-snodgrass) [inline server/react-admin/src/AdminApp.js:26]: we should default this to false so it doesn't flash open when navigating back when using the browser back button

File: server/react-admin/src/sidemenu/SideNav.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        <SideNavTab label="Geographies" link="geospatialDefinitions" />
+        <SideNavTab label="Officials" link="officials" />
+        <SideNavTab label="Publications" link="publications" />
+        <SideNavTab
+          label="Video Testimonials"
+          link="videoTestimonials"
+          component="Video Testimonial"
+        />
+        <SideNavTab label="Legislative Artifacts" link="legislativeArtifacts" />
+        <SideNavTab label="API Docs" link="api-docs" />
```
Comment (ed-snodgrass) [inline server/react-admin/src/sidemenu/SideNav.js:84]: I think we could save a lot of css by doing something like
```
        <List>
          {navItems.map((navItem) => (
            <ListItemButton key={navItem.label} href={navItem.link} component="a">
              <ListItemText primary={navItem.label} />
            </ListItemButton>
          ))}
        </List>
```
where `navItems` is
```
const navItems = [
  { label: 'Legislative Artifacts', link: 'legislativeArtifacts' },
  { label: 'Advocacy Groups', link: 'advocacyGroups' },
  { label: 'Categories', link: 'categories' },
  { label: 'Levels', link: 'levels' },
  { label: 'Geographies', link: 'geospatialDefinitions' },
  { label: 'Officials', link: 'officials' },
  { label: 'Publications', link: 'publications' },
  { label: 'Video Testimonials', link: 'videoTestimonials' },
  { label: 'API Docs', link: 'api-docs' },
]
```

**Comment by ed-snodgrass** [comment]
> This looks great. I'd like to get this on main so we have the functionality, but first I'm hoping @ed-snodgrass has a chance to look in particular to make sure he agrees with the BrowserRouter rearranging. So we don't go back and forth on something like that (unless back and forth good iterations). I don't see a problem with it.

I'm not real sure that I have an opinion on the moving of the BrowserRouter. I'm good with pulling it in, even if you do want to keep the expanded menu items with the separate list and add/edit views.

**Comment by markstur** [comment]
> @markstur I'm curious if we could simplify the side menu a bit and just have the links to the individual components? The table components themselves have the add/edit options.
> 
> Something like: image

Yes. I had the same thought when looking at this PR. I probably only had them separate because that is how the thing evolved and then just kept it to show me it was complete (or maybe just because I already had expandable items in the menu?).  For usability, it sounds like you and I agree that it'd be better to just have the entities in the sidebar and go straight to the list view which already has the add button.

**Comment by markstur** [comment]
See Ed's comments.  For the simplified sidebar, that could be a separate PR if you want to address these things first. Up to you.

**Review by ed-snodgrass — APPROVED** [review]
looks good!

**Pull request merged**